Fix deserialization with msgpack 1.0.

Patch copied from upstream source repository:
https://github.com/kovidgoyal/calibre/commit/0ff41ac64994ec11b7859fc004c94d08769e3af3

diff --git a/src/calibre/utils/serialize.py b/src/calibre/utils/serialize.py
index f5d560c468..c35ae53849 100644
--- a/src/calibre/utils/serialize.py
+++ b/src/calibre/utils/serialize.py
@@ -110,7 +110,7 @@ def msgpack_decoder(code, data):
 def msgpack_loads(dump, use_list=True):
     # use_list controls whether msgpack arrays are unpacked as lists or tuples
     import msgpack
-    return msgpack.unpackb(dump, ext_hook=msgpack_decoder, raw=False, use_list=use_list)
+    return msgpack.unpackb(dump, ext_hook=msgpack_decoder, raw=False, use_list=use_list, strict_map_key=False)
 
 
 def json_loads(data):
